test/views/html/TaxCalculationViewSpec.scala

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.html
+
+import models._
+import org.joda.time.LocalDate
+import org.jsoup.nodes.Document
+import views.html.cards.home.taxCalculation
+import models.UnderpaidStatus.{Unknown => UnderpaidUnknown, _}
+import models.OverpaidStatus.{Unknown => OverpaidUnknown, _}
+import util.{BetterOptionValues, LanguageHelper}
+import viewmodels.TaxCalculationViewModel
+
+class TaxCalculationViewSpec extends ViewSpec {
+
+  import BetterOptionValues._
+
+  def view(reconciliation: Reconciliation): Option[Document] =
+    TaxCalculationViewModel.fromTaxYearReconciliation(TaxYearReconciliation(2017, reconciliation))(config).map {
+      taxRec =>
+        asDocument(taxCalculation(taxRec)(messages, pertaxContext, config).toString)
+    }
+
+  def formatDate(date: LocalDate) = LanguageHelper.langUtils.Dates.formatDate(Some(date), "dd MMMM yyyy")(messages)
+
```

Comment:
I think a lot of the logic being tested here overlaps with the tests in HomeCardGeneratorSpec, could simplify this spec after the extraction of view logic into one place.

Reply:
I would argue that these tests are checking that the correct links and actual content are being rendered as expected whereas the HomeCardGeneratorSpec checks that the function which calls the view behaves as expected by returning HTML given certain pre-conditions and circumstances.